### PR TITLE
Fixes #10152 - reduce num queries in sys#index

### PR DESCRIPTION
**Pull request by dustints**
Fixes #10152 - reduce num querys in sys#index …
* system's rabl  index renders without the use of the show template
* include :content_view, :activation_keys in elastic query for N+1 issue

### CODE REVIEW

**Comment by dustints** [comment]
`curl http://katello:3000/katello/api/v2/systems\?organization_id\=11\&page\=1\&search\=\&sort_by\=name\&sort_order\=ASC\&minimal\=true --basic -u admin:changeme  | python -m json.tool; tput bel`
on a 11 host organization
this change seemed to reduce the number for queries from 352 to 79

File: app/controllers/katello/api/v2/systems_controller.rb [hunk 1]
```diff
@@ -69,18 +68,45 @@ def organization_id_keys
...
+      #before filters
+      m = Benchmark.measure do
+        find_environment
+        find_optional_organization
+        find_host_collection
+        find_default_organization_and_or_environment
+        load_search_service
+      end
+      Rails.logger.error('xxxxbefore filter bench')
+      Rails.logger.error(m)
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/systems_controller.rb]: left some debugging :)

File: app/views/katello/api/v2/systems/index.json.rabl [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+child @collection[:results] => :results do
+  attributes :id => :katello_id
+  attributes :uuid, :name, :description
+  attributes :location
+  attributes :content_view, :content_view_id
+  attributes :distribution
+  attributes :katello_agent_installed? => :katello_agent_installed
+  attributes :registered_by
+end
```
Comment (jlsherrill) [inline app/views/katello/api/v2/systems/index.json.rabl]: we've used a paradigm of base.json.rabl to define 'common' attributes, and then show and index simply extending that.  See content_view_version rabl templates for an example.  I'd like to see that here

**Comment by dustints** [comment]
benchmarking patch for pr tester.
https://gist.github.com/anonymous/5cdb6479e6602419f70a

**Comment by dustints** [comment]
@jlsherrill updated.
also tested loading from the web ui...
for 70 content hosts on the content-host index page, queries were reduced from 622 to 51

File: app/views/katello/api/v2/systems/base.json.rabl [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+@resource ||= @object
+
+node(:id) { |resource| resource.uuid }
+attributes :id => :katello_id
+attributes :uuid, :name, :description
+attributes :location
+attributes :content_view, :content_view_id
+attributes :distribution
+attributes :katello_agent_installed? => :katello_agent_installed
+attributes :registered_by
```
Comment (jlsherrill) [inline app/views/katello/api/v2/systems/base.json.rabl]: Looks like some needed fields are missing in the base rabl:

lifecycle environment, registered (time), last checkin, errata counts

In the UI these items are blank

**Comment by dustints** [comment]
@jlsherrill updated, think I believe i was missing environment and activation_keys...there's some view logic that   displays the registration info.

**Comment by dustints** [comment]
@jlsherrill should be good now. ack?

**Comment by jlsherrill** [comment]
@dustints still missing errata counts

**Comment by jlsherrill** [comment]
ACK, opened http://projects.theforeman.org/issues/10497 for further improvements

**Pull request merged**